### ISSUE TITLE
Not able to prepare the dataset shape correctly.

### ISSUE THREAD

**Issue by hiroshiperera** [Sunnydreamrain/IndRNN_pytorch#3]
Hi ... Thanks alot for this fabulous work. I'm still trying to understand the code and and trying to run this in my machine.
The main issue I'm having is, I'm not able to prepare the dataset as I required.

I'm trying to understand the following code

datasets=train_datasets
dataname=datasets+'.npy'
labelname=datasets+'_label.npy'
lenname=datasets+'_len.npy'
data_handle=np.load(dataname)
label_handle=np.load(labelname)
len_handle=np.load(lenname)

What is this train_datasets should be? What's it shape need to be? Can you elaborate more on this. Because I need to arrange it to feed to the network

**Comment by Sunnydreamrain** [comment]
Hi,   
First, as mentioned, you can always use your own data loader.
The shape of the dataset I prepared is (samples, time, joints, dimension), which should be something like (38086, 300, 50, 3). '300' means the length of the sequence, where the true length is indicated in the file lenname=datasets+'_len.npy'. '50' is the number of joints and '3' is the xyz dimensions.   

It should also be able to be derived from the relevant code.  

data_handle=np.load(dataname)  line 23  
dataset=data_handle[shufflevideoindex]  line 79  
sample=dataset[startid:startid+self.seq_len]  line 86

**Comment by hiroshiperera** [comment]
Thanks alot for your answer.. I'm having few more questions. Sorry I'm still in the learning phase. when considers '50' the maximum number of joints in a given time frame is 25. But since it is a 3D shaped image, in total there are 75 (25x3) columns. In your example may I know why you took it as 50. Is it like (25x2), which means 50 joints are feeding to the network? Does 300 means 300 data rows are feeding to the network. Here I elaborate it in the following image in more details. Does 38086 means total number of rows taking as the sample size? Following image shows the .mat file of a particular person. So to make 38086, we would combine the time frames of multiple people and creating a single dataframe? In the following image there were 4450 rows. By combining more .mat files, I'm planning to make a large sample. Can you correct if I'm wrong.

![image](https://user-images.githubusercontent.com/6210895/57219053-90a68600-7014-11e9-9acd-41a558bc57ea.png)

**Comment by Sunnydreamrain** [comment]
Hi,

50 joints meaning two persons with each person having 25 joints.

**Comment by hiroshiperera** [comment]
Thanks alot :)

**Comment by hiroshiperera** [comment]
Hi, Is there any possibility to attach your dataname=datasets+'.npy'
labelname=datasets+'_label.npy'
lenname=datasets+'_len.npy'

files here? I really appreciate if it is possible. And also I'm not clear what the lenname array looks like. Kindly help

**Comment by Sunnydreamrain** [comment]
Hi,  
The file is too large.  
As I mentioned repeatedly, you don't have to use my data loader. You can just write one of your own according to you understanding.  
datasets+'_len.npy' is used because the lengths of different samples are different.  
datasets+'_label.npy' apparently are the labels.

**Comment by WangMuT** [comment]
Hi,
Thank you verymuch for sharing your knowledge.
I downloaded the NTU skeleton dataset and the files are in .skeleton format.
Did you convert that .skeleton files to np array inorder to make inputs ?
Thank you.

**Comment by Sunnydreamrain** [comment]
Hi, @WangMuT 

Yes.
